client/state/reader/conversations/test/reducer.js

```diff
@@ -56,6 +57,54 @@ describe( 'reducer', () => {
 			expect( state[ '123-456' ] ).toEqual( 'F' );
 		} );
 
+		test( 'should remove the given site and post from state entirely if followStatus is null', () => {
+			const original = deepFreeze( { '123-456': 'M' } );
+
+			const state = items( original, {
+				type: READER_CONVERSATION_UPDATE_FOLLOW_STATUS,
+				payload: { siteId: 123, postId: 456, followStatus: null },
+			} );
+
+			expect( state[ '123-456' ] ).toEqual( undefined );
+		} );
+
+		test( 'should add a new follow when new posts are received', () => {
+			const original = deepFreeze( {} );
+
+			const state = items( original, {
+				type: READER_POSTS_RECEIVE,
+				posts: [
+					{ site_ID: 123, ID: 456, is_following_conversation: true },
+					{ site_ID: 123, ID: 789, is_following_conversation: true },
+				],
+			} );
+
+			expect( state[ '123-456' ] ).toEqual( 'F' );
+			expect( state[ '123-789' ] ).toEqual( 'F' );
+		} );
+
+		test( 'should not a new follow when new posts are received if is_following_conversation is false', () => {
+			const original = deepFreeze( {} );
+
+			const state = items( original, {
+				type: READER_POSTS_RECEIVE,
+				posts: [ { site_ID: 123, ID: 456, is_following_conversation: false } ],
+			} );
+
+			expect( state[ '123-456' ] ).toEqual( undefined );
```

Comment:
technically we want to verify that the key doesn't even exist -- not just that it is set to undefined.

After looking at jest docs it seems like we want: `.not` and `.toHaveProperty`
```js
expect( state ).not().toHaveProperty( '123-456' );
```

Reply:
👍 It's:

`expect( state ).not.toHaveProperty( '123-456' );`